common/step_download.go

```diff
@@ -61,6 +62,10 @@ func (s *StepDownload) Run(state multistep.StateBag) multistep.StepAction {
 
 	ui.Say(fmt.Sprintf("Downloading or copying %s", s.Description))
 
+	// Get a default-looking progress bar and connect it to the ui.
```

Comment:
I'm not sold on the progress bar and changes to the download code for a really simple reason -- it's a nice-to-have and therefore less urgent than your DownloadableURL changes, and I'd like to separate bugfixes from enhancements. I think I like the idea of a progress bar but if you're game I'd like to separate it out into a different PR so we have the luxury of going over it separately. And hopefully come to a conclusion on it sometime before two more years pass 😬

Reply:
Yea. Again I added the progress bar since the way that I added the support for the other protocols was building on mitchellh's code and making them all accessible asynchronously via goroutines. Personally I kind of hate chegggaa's progressbar but it was the one that some users chose when I first wrote this patch.

I can split it into another branch, just it'll be dependent on this PR. I'll try and get to these splits this weekend.